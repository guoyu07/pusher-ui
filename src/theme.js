const fontFamilies = [
  '-apple-system',
  'BlinkMacSystemFont',
  'Segoe UI',
  'Roboto',
  'Oxygen',
  'Ubuntu',
  'Cantarell',
  'Fira Sans',
  'Droid Sans',
  'Helvetica Neue',
  'sans-serif',
].map(name => `"${name}"`).join(', ');


export const fontFamily = fontFamilies;
export const white = '#fff';
export const black = '#1e1e1e';
export const grey = '#e7ebf3';
export const darkGrey = '#d6d6d6';
export const darkDarkGrey = '#989898';
export const lightGrey = '#e7ebf3';
export const primary = '#549eff';
export const success = '#319a31';
export const warning = '#ffaf1c';
export const danger = '#ca0808';
export const zIndex1 = 1;
export const zIndex2 = 2;
export default {
  fontFamily,
  white,
  black,
  grey,
  darkGrey,
  darkDarkGrey,
  lightGrey,
  primary,
  success,
  warning,
  danger,
  zIndex1,
  zIndex2,
};
