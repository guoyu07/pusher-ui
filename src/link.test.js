import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'glamorous';

import { Link } from './index';
import * as theme from './theme';

describe('Link', () => {
  const defaultProps = {
    to: 'https://example.com',
  };
  const createSnapshot = (props = defaultProps) => {
    expect(
      renderer
        .create(
          <ThemeProvider theme={theme}>
            <Link {...props} />
          </ThemeProvider>,
        )
        .toJSON(),
    ).toMatchSnapshot();
  };

  it('should render a link', () => {
    createSnapshot();
  });

  it('should pass on any css props', () => {
    createSnapshot({
      ...defaultProps,
      padding: 24,
      margin: 12,
      color: 'hotpink',
    });
  });
});
