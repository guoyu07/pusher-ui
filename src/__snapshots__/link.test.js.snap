// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should pass on any css props 1`] = `
.glamor-0,
[data-glamor-0] {
  text-decoration: none;
  color: #006eff;
}

.glamor-0:hover,
[data-glamor-0]:hover {
  text-decoration: underline;
  cursor: pointer;
}

<a
  className="glamor-0"
  color="hotpink"
  margin={12}
  onClick={[Function]}
  padding={24}
  style={Object {}}
/>
`;

exports[`Link should render a link 1`] = `
.glamor-0,
[data-glamor-0] {
  text-decoration: none;
  color: #006eff;
}

.glamor-0:hover,
[data-glamor-0]:hover {
  text-decoration: underline;
  cursor: pointer;
}

<a
  className="glamor-0"
  onClick={[Function]}
  style={Object {}}
/>
`;
